From 899d82225249bf78a4a3544926ead8ec604236cc Mon Sep 17 00:00:00 2001
From: Qi Feng <qi.feng@arm.com>
Date: Wed, 18 Oct 2023 16:37:19 +0800
Subject: [PATCH] efi: vexpress64: Enable capsule update, get_image_info and
 esrt

This commit enables the following UEFI services/features:

* Capsule Update
* GetImageInfo
* ESRT table

Fixes the following errors:

* doc format error raised by scripts/kernel-doc.
* ../../drivers/mtd/nvmxip/nvmxip.h not found error.
* undefined reference to `num_image_type_guids'/`update_info'

Also rename the symbol names to common ones to reflect the fact, that
vexpress64 and corstone1000 use common functions and variables.

Upstream-Status: Pending
Signed-off-by: Qi Feng <qi.feng@arm.com>
Signed-off-by: Robbie Cao <robbie.cao@arm.com>
---
 board/armltd/vexpress64/vexpress64.c |  4 ++
 configs/corstone1000_defconfig       |  1 +
 include/configs/corstone1000.h       | 10 ++---
 include/configs/vexpress_aemv8.h     | 20 ++++++++++
 lib/efi_loader/efi_boottime.c        |  2 +-
 lib/efi_loader/efi_capsule.c         | 55 +++++++++++++---------------
 lib/efi_loader/efi_firmware.c        | 38 ++++++++++---------
 lib/efi_loader/efi_setup.c           | 34 ++++++++++-------
 test/dm/nvmxip.c                     |  2 +-
 9 files changed, 99 insertions(+), 67 deletions(-)

diff --git a/board/armltd/vexpress64/vexpress64.c b/board/armltd/vexpress64/vexpress64.c
index 9b84dd8c7c..33728f63aa 100644
--- a/board/armltd/vexpress64/vexpress64.c
+++ b/board/armltd/vexpress64/vexpress64.c
@@ -94,6 +94,10 @@ __weak void vexpress64_pcie_init(void)
 {
 }
 
+void set_dfu_alt_info(char *interface, char *devstr)
+{
+}
+
 int board_init(void)
 {
 	vexpress64_pcie_init();
diff --git a/configs/corstone1000_defconfig b/configs/corstone1000_defconfig
index a0af413de8..a29c7f05b1 100644
--- a/configs/corstone1000_defconfig
+++ b/configs/corstone1000_defconfig
@@ -60,6 +60,7 @@ CONFIG_USB_ISP1760=y
 CONFIG_ERRNO_STR=y
 CONFIG_NVMXIP_QSPI=y
 CONFIG_EFI_MM_COMM_TEE=y
+CONFIG_ARM_FFA_TRANSPORT=y
 CONFIG_FFA_SHARED_MM_BUF_SIZE=4096
 CONFIG_FFA_SHARED_MM_BUF_OFFSET=0
 CONFIG_FFA_SHARED_MM_BUF_ADDR=0x02000000
diff --git a/include/configs/corstone1000.h b/include/configs/corstone1000.h
index 8622565a87..1c01a07067 100644
--- a/include/configs/corstone1000.h
+++ b/include/configs/corstone1000.h
@@ -15,14 +15,14 @@
 #include <linux/sizes.h>
 
 /* The SE Proxy partition ID*/
-#define CORSTONE1000_SEPROXY_PART_ID		(0x8002)
+#define FVP_SEPROXY_PART_ID		(0x8002)
 
 /* Update service ID provided by  the SE Proxy secure partition*/
-#define CORSTONE1000_SEPROXY_UPDATE_SVC_ID	(0x4)
+#define FVP_SEPROXY_UPDATE_SVC_ID	(0x4)
 
 /* Notification events used with SE Proxy update service */
-#define CORSTONE1000_BUFFER_READY_EVT		(0x1)
-#define CORSTONE1000_UBOOT_EFI_STARTED_EVT	(0x2)
+#define FVP_BUFFER_READY_EVT		(0x1)
+#define FVP_UBOOT_EFI_STARTED_EVT	(0x2)
 
 #define PREP_SEPROXY_SVC_ID_MASK	GENMASK(31, 16)
 #define PREP_SEPROXY_SVC_ID(x)	 (FIELD_PREP(PREP_SEPROXY_SVC_ID_MASK, (x)))
@@ -31,7 +31,7 @@
 #define PREP_SEPROXY_EVT(x)	(FIELD_PREP(PREP_SEPROXY_EVT_MASK, (x)))
 
 /* Size in 4KB pages of the EFI capsule buffer */
-#define CORSTONE1000_CAPSULE_BUFFER_SIZE	(8192) /* 32 MB */
+#define FVP_CAPSULE_BUFFER_SIZE	(8192) /* 32 MB */
 
 /* Capsule GUID */
 #define EFI_CORSTONE1000_CAPSULE_ID_GUID \
diff --git a/include/configs/vexpress_aemv8.h b/include/configs/vexpress_aemv8.h
index 24d8ca0866..c1100d257d 100644
--- a/include/configs/vexpress_aemv8.h
+++ b/include/configs/vexpress_aemv8.h
@@ -9,6 +9,26 @@
 
 #include <linux/stringify.h>
 
+
+/* The SE Proxy partition ID*/
+#define FVP_SEPROXY_PART_ID		(0x8002)
+
+/* Update service ID provided by  the SE Proxy secure partition*/
+#define FVP_SEPROXY_UPDATE_SVC_ID	(0x4)
+
+/* Notification events used with SE Proxy update service */
+#define FVP_BUFFER_READY_EVT		(0x1)
+#define FVP_UBOOT_EFI_STARTED_EVT	(0x2)
+
+#define PREP_SEPROXY_SVC_ID_MASK	GENMASK(31, 16)
+#define PREP_SEPROXY_SVC_ID(x)	 (FIELD_PREP(PREP_SEPROXY_SVC_ID_MASK, (x)))
+
+#define PREP_SEPROXY_EVT_MASK		GENMASK(15, 0)
+#define PREP_SEPROXY_EVT(x)	(FIELD_PREP(PREP_SEPROXY_EVT_MASK, (x)))
+
+/* Size in 4KB pages of the EFI capsule buffer */
+#define FVP_CAPSULE_BUFFER_SIZE	(8192) /* 32 MB */
+
 /* Link Definitions */
 #ifdef CONFIG_TARGET_VEXPRESS64_JUNO
 #else
diff --git a/lib/efi_loader/efi_boottime.c b/lib/efi_loader/efi_boottime.c
index a5da77c36c..6ea0e4cf3d 100644
--- a/lib/efi_loader/efi_boottime.c
+++ b/lib/efi_loader/efi_boottime.c
@@ -23,7 +23,7 @@
 #include <asm/setjmp.h>
 #include <linux/libfdt_env.h>
 
-#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 #include <arm_ffa.h>
 #include <dm.h>
 #include <linux/bitfield.h>
diff --git a/lib/efi_loader/efi_capsule.c b/lib/efi_loader/efi_capsule.c
index 676acefa8e..577c64926d 100644
--- a/lib/efi_loader/efi_capsule.c
+++ b/lib/efi_loader/efi_capsule.c
@@ -26,22 +26,19 @@
 #include <crypto/pkcs7_parser.h>
 #include <linux/err.h>
 
-#ifdef CONFIG_TARGET_CORSTONE1000
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 #include <arm_ffa.h>
 #include <cpu_func.h>
 #include <dm.h>
 #include <linux/bitfield.h>
 #include <linux/bitops.h>
 
-void *__efi_runtime_data corstone1000_capsule_buf; /* capsule shared buffer virtual address */
-efi_guid_t corstone1000_capsule_guid = EFI_CORSTONE1000_CAPSULE_ID_GUID;
+void *__efi_runtime_data fvp_capsule_buf; /* capsule shared buffer virtual address */
 #endif
 
 DECLARE_GLOBAL_DATA_PTR;
 
 const efi_guid_t efi_guid_capsule_report = EFI_CAPSULE_REPORT_GUID;
-static const efi_guid_t efi_guid_firmware_management_capsule_id =
-		EFI_FIRMWARE_MANAGEMENT_CAPSULE_ID_GUID;
 const efi_guid_t efi_guid_firmware_management_protocol =
 		EFI_FIRMWARE_MANAGEMENT_PROTOCOL_GUID;
 const efi_guid_t fwu_guid_os_request_fw_revert =
@@ -720,10 +717,10 @@ static efi_status_t efi_capsule_update_firmware(
 }
 #endif /* CONFIG_EFI_CAPSULE_FIRMWARE_MANAGEMENT */
 
-#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 
 /**
- * efi_corstone1000_alloc_capsule_shared_buf - allocate capsule shared buffer
+ * efi_fvp_alloc_capsule_shared_buf - allocate capsule shared buffer
  * @capsule_image_size:	The capsule data (header + payload)
  *
  * This function allocates the physically contiguous buffer shared between u-boot
@@ -736,7 +733,7 @@ static efi_status_t efi_capsule_update_firmware(
  *
  * 0: on success, otherwise failure
  */
-efi_status_t efi_corstone1000_alloc_capsule_shared_buf(void)
+efi_status_t efi_fvp_alloc_capsule_shared_buf(void)
 {
 	efi_status_t efi_ret;
 	u64 ram_base = CFG_SYS_SDRAM_BASE;
@@ -745,27 +742,27 @@ efi_status_t efi_corstone1000_alloc_capsule_shared_buf(void)
 
 	efi_ret = efi_allocate_pages(EFI_ALLOCATE_ADDRESS,
 				     EFI_RUNTIME_SERVICES_DATA,
-				     CORSTONE1000_CAPSULE_BUFFER_SIZE,
+				     FVP_CAPSULE_BUFFER_SIZE,
 				     &ram_base);
 
 	if (efi_ret != EFI_SUCCESS) {
-		corstone1000_capsule_buf = NULL;
-		log_err("EFI: Corstone1000: Allocating capsule shared buffer error (%d)\n"
+		fvp_capsule_buf = NULL;
+		log_err("EFI: FVP: Allocating capsule shared buffer error (%d)\n"
 			, (int)efi_ret);
 		return efi_ret;
 	}
 
-	log_info("EFI: Corstone1000: Capsule shared buffer at 0x%x , size %d pages\n"
+	log_info("EFI: FVP: Capsule shared buffer at 0x%x , size %d pages\n"
 		, (unsigned int)ram_base,
-		CORSTONE1000_CAPSULE_BUFFER_SIZE);
+		FVP_CAPSULE_BUFFER_SIZE);
 
-	corstone1000_capsule_buf = (void *)map_sysmem((phys_addr_t)ram_base, 0);
+	fvp_capsule_buf = (void *)map_sysmem((phys_addr_t)ram_base, 0);
 
 	return EFI_SUCCESS;
 }
 
 /**
- * efi_corstone1000_buffer_ready_event - issue door bell event
+ * efi_fvp_buffer_ready_event - issue door bell event
  * @capsule_image_size:	The capsule data (header + payload)
  *
  * This function notifies the SE Proxy update service that capsule data is available
@@ -775,7 +772,7 @@ efi_status_t efi_corstone1000_alloc_capsule_shared_buf(void)
  *
  * 0: on success, otherwise failure
  */
-static int __efi_runtime efi_corstone1000_buffer_ready_event(u32 capsule_image_size)
+static int __efi_runtime efi_fvp_buffer_ready_event(u32 capsule_image_size)
 {
 	struct ffa_send_direct_data msg = {0};
 	int ret;
@@ -794,10 +791,10 @@ static int __efi_runtime efi_corstone1000_buffer_ready_event(u32 capsule_image_s
 	 * - capsule update interface ID (31:16)
 	 * - the buffer ready event ID (15:0)
 	 */
-	msg.data1 = PREP_SEPROXY_SVC_ID(CORSTONE1000_SEPROXY_UPDATE_SVC_ID) |
-		PREP_SEPROXY_EVT(CORSTONE1000_BUFFER_READY_EVT); /* w4 */
+	msg.data1 = PREP_SEPROXY_SVC_ID(FVP_SEPROXY_UPDATE_SVC_ID) |
+		PREP_SEPROXY_EVT(FVP_BUFFER_READY_EVT); /* w4 */
 
-	return ffa_sync_send_receive(dev, CORSTONE1000_SEPROXY_PART_ID, &msg, 0);
+	return ffa_sync_send_receive(dev, FVP_SEPROXY_PART_ID, &msg, 0);
 }
 #endif
 
@@ -831,8 +828,8 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 		goto out;
 	}
 
-#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
-		if (capsule_count != 1 || !corstone1000_capsule_buf) {
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
+		if (capsule_count != 1 || !fvp_capsule_buf) {
 			ret = EFI_INVALID_PARAMETER;
 			goto out;
 		}
@@ -851,24 +848,24 @@ efi_status_t __efi_runtime EFIAPI efi_update_capsule(
 		log_debug("Capsule[%d] (guid:%pUs)\n",
 			  i, &capsule->capsule_guid);
 
-#if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 		if (efi_size_in_pages(capsule->capsule_image_size) >
-		    CORSTONE1000_CAPSULE_BUFFER_SIZE) {
-			log_err("Corstone1000: Capsule data size exceeds the shared buffer size\n");
+		    FVP_CAPSULE_BUFFER_SIZE) {
+			log_err("FVP: Capsule data size exceeds the shared buffer size\n");
 			ret = EFI_BUFFER_TOO_SMALL;
 			goto out;
 		}
 
 		/* copy the data to the contiguous buffer */
-		efi_memcpy_runtime(corstone1000_capsule_buf, capsule, capsule->capsule_image_size);
+		efi_memcpy_runtime(fvp_capsule_buf, capsule, capsule->capsule_image_size);
 
 		/* invalidate the data cache */
 		invalidate_dcache_all();
 
 		/* issue buffer ready event */
-		ret = efi_corstone1000_buffer_ready_event(capsule->capsule_image_size);
+		ret = efi_fvp_buffer_ready_event(capsule->capsule_image_size);
 		if (ret) {
-			log_err("EFI: Corstone1000: Buffer ready event error (%d)\n", (int)ret);
+			log_err("EFI: FVP: Buffer ready event error (%d)\n", (int)ret);
 			ret = EFI_DEVICE_ERROR;
 		} else {
 			ret = EFI_SUCCESS;
@@ -1460,8 +1457,8 @@ efi_status_t efi_launch_capsules(void)
 			index = 0;
 		ret = efi_capsule_read_file(files[i], &capsule);
 		if (ret == EFI_SUCCESS) {
-			#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
-			/* capsule update only supports 1 image and no scatter gather list for corstone1000 */
+			#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
+			/* capsule update only supports 1 image and no scatter gather list for corstone1000 and vexpress64 */
 			efi_update_capsule(&capsule, 1, 0);
 			#else
 			ret = efi_capsule_update_firmware(capsule);
diff --git a/lib/efi_loader/efi_firmware.c b/lib/efi_loader/efi_firmware.c
index c6ab6e2182..9d6c475f1d 100644
--- a/lib/efi_loader/efi_firmware.c
+++ b/lib/efi_loader/efi_firmware.c
@@ -19,7 +19,7 @@
 
 #define FMP_PAYLOAD_HDR_SIGNATURE	SIGNATURE_32('M', 'S', 'S', '1')
 
-#if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 #define EFI_FIRMWARE_IMAGE_TYPE_UBOOT_RAW_GUID \
 	EFI_GUID(0xe2bb9c06, 0x70e9, 0x4b14, 0x97, 0xa3, \
 		 0x5a, 0x79, 0x13, 0x17, 0x6e, 0x3f)
@@ -27,7 +27,7 @@
  const efi_guid_t efi_firmware_image_type_uboot_raw =
 				EFI_FIRMWARE_IMAGE_TYPE_UBOOT_RAW_GUID;
 
-static efi_status_t efi_corstone1000_img_info_get (
+static efi_status_t efi_fvp_img_info_get (
 	efi_uintn_t *image_info_size,
 	struct efi_firmware_image_descriptor *image_info,
 	u32 *descriptor_version,
@@ -164,6 +164,7 @@ efi_status_t EFIAPI efi_firmware_set_package_info_unsupported(
 	return EFI_EXIT(EFI_UNSUPPORTED);
 }
 
+#if !(IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP))
 /**
  * efi_fill_image_desc_array - populate image descriptor array
  * @image_info_size:		Size of @image_info
@@ -239,6 +240,7 @@ static efi_status_t efi_fill_image_desc_array(
 
 	return EFI_SUCCESS;
 }
+#endif
 
 /**
  * efi_firmware_capsule_authenticate - authenticate the capsule if enabled
@@ -304,6 +306,7 @@ efi_status_t efi_firmware_capsule_authenticate(const void **p_image,
 	return EFI_SUCCESS;
 }
 
+#if !IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
 /**
  * efi_firmware_get_image_info - return information about the current
  *				     firmware image
@@ -348,8 +351,8 @@ efi_status_t EFIAPI efi_firmware_get_image_info(
 	     !descriptor_size || !package_version || !package_version_name))
 		return EFI_EXIT(EFI_INVALID_PARAMETER);
 
-#if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
-	ret = efi_corstone1000_img_info_get(image_info_size, image_info,
+#if IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
+	ret = efi_fvp_img_info_get(image_info_size, image_info,
 			       descriptor_version, descriptor_count,
 			       descriptor_size,
 			       package_version, package_version_name,
@@ -363,6 +366,7 @@ efi_status_t EFIAPI efi_firmware_get_image_info(
 
 	return EFI_EXIT(ret);
 }
+#endif
 
 #ifdef CONFIG_EFI_CAPSULE_FIRMWARE_FIT
 /*
@@ -418,19 +422,19 @@ efi_status_t EFIAPI efi_firmware_fit_set_image(
 	return EFI_EXIT(EFI_SUCCESS);
 }
 
-#if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 
 /**
- * efi_firmware_corstone1000_get_image_info - return information about the current
-                                    firmware image
+ * efi_firmware_fvp_get_image_info - return information about the current
+ *                                  firmware image
  * @this:                      Protocol instance
  * @image_info_size:           Size of @image_info
- * @image_info:                        Image information
- * @descriptor_version:                Pointer to version number
+ * @image_info:                Image information
+ * @descriptor_version:        Pointer to version number
  * @descriptor_count:          Pointer to number of descriptors
  * @descriptor_size:           Pointer to descriptor size
- * package_version:            Package version
- * package_version_name:       Package version's name
+ * @package_version:           Package version
+ * @package_version_name:      Package version's name
  *
  * Return information bout the current firmware image in @image_info.
  * @image_info will consist of a number of descriptors.
@@ -439,7 +443,7 @@ efi_status_t EFIAPI efi_firmware_fit_set_image(
  * Return              status code
  */
 static
-efi_status_t EFIAPI efi_firmware_corstone1000_get_image_info(
+efi_status_t EFIAPI efi_firmware_fvp_get_image_info(
        struct efi_firmware_management_protocol *this,
        efi_uintn_t *image_info_size,
        struct efi_firmware_image_descriptor *image_info,
@@ -523,7 +527,7 @@ efi_status_t EFIAPI efi_firmware_corstone1000_get_image_info(
 	       return EFI_EXIT(ret);
 
 	image_info_version_size_var -= image_info_name_size_var;
-	image_info->image_id_name = runner;
+	image_info->image_id_name = (u16*)runner;
 	runner += image_info_name_size_var;
 
         /* Consider changing the string modfication logic */
@@ -534,7 +538,7 @@ efi_status_t EFIAPI efi_firmware_corstone1000_get_image_info(
 	if (ret != EFI_SUCCESS)
 	        return EFI_EXIT(ret);
 
-	image_info->version_name = runner;
+	image_info->version_name = (u16*)runner;
 
 	*image_info_size = image_info_size_var;
 
@@ -547,8 +551,8 @@ efi_status_t EFIAPI efi_firmware_corstone1000_get_image_info(
 #endif
 
 const struct efi_firmware_management_protocol efi_fmp_fit = {
-#if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
-	.get_image_info = efi_firmware_corstone1000_get_image_info,
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
+	.get_image_info = efi_firmware_fvp_get_image_info,
 #else
 	.get_image_info = efi_firmware_get_image_info,
 #endif
@@ -617,7 +621,7 @@ efi_status_t EFIAPI efi_firmware_raw_set_image(
 		}
 	}
 
-#if CONFIG_IS_ENABLED(TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 	return EFI_EXIT(EFI_SUCCESS);
 #endif
 
diff --git a/lib/efi_loader/efi_setup.c b/lib/efi_loader/efi_setup.c
index d20568c1c8..996982c331 100644
--- a/lib/efi_loader/efi_setup.c
+++ b/lib/efi_loader/efi_setup.c
@@ -17,16 +17,16 @@
 
 efi_status_t efi_obj_list_initialized = OBJ_LIST_NOT_INITIALIZED;
 
-#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000) || IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP)
 #include <linux/bitfield.h>
 #include <linux/bitops.h>
 #include <arm_ffa.h>
 #include <dm.h>
 
 /**
- * efi_corstone1000_alloc_capsule_shared_buf - allocate capsule shared buffer
+ * efi_fvp_alloc_capsule_shared_buf - allocate capsule shared buffer
  */
-extern efi_status_t efi_corstone1000_alloc_capsule_shared_buf(void);
+extern efi_status_t efi_fvp_alloc_capsule_shared_buf(void);
 #endif
 
 /*
@@ -132,9 +132,11 @@ static efi_status_t efi_init_secure_boot(void)
 }
 #endif /* CONFIG_EFI_SECURE_BOOT */
 
-#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_EFI_HAVE_CAPSULE_SUPPORT) &&	\
+	(IS_ENABLED(CONFIG_TARGET_CORSTONE1000) ||	\
+	 IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP))
 /**
- * efi_corstone1000_uboot-efi_started_event - notifies SE Proxy FW update service
+ * efi_fvp_uboot_efi_started_event - notifies SE Proxy FW update service
  *
  * This function notifies the SE Proxy update service that uboot efi has already started
  *
@@ -142,7 +144,7 @@ static efi_status_t efi_init_secure_boot(void)
  *
  * 0: on success, otherwise failure
  * */
-static int efi_corstone1000_uboot_efi_started_event(void)
+static int efi_fvp_uboot_efi_started_event(void)
 {
 	struct ffa_send_direct_data msg = {0};
 	int ret;
@@ -161,10 +163,10 @@ static int efi_corstone1000_uboot_efi_started_event(void)
 	 * setting capsule update interface ID(31:16)
 	 * the kernel started event ID(15:0)
 	 */
-	msg.data1 = PREP_SEPROXY_SVC_ID(CORSTONE1000_SEPROXY_UPDATE_SVC_ID) |
-			PREP_SEPROXY_EVT(CORSTONE1000_UBOOT_EFI_STARTED_EVT); /* w4 */
+	msg.data1 = PREP_SEPROXY_SVC_ID(FVP_SEPROXY_UPDATE_SVC_ID) |
+			PREP_SEPROXY_EVT(FVP_UBOOT_EFI_STARTED_EVT); /* w4 */
 
-	return ffa_sync_send_receive(dev, CORSTONE1000_SEPROXY_PART_ID, &msg, 0);
+	return ffa_sync_send_receive(dev, FVP_SEPROXY_PART_ID, &msg, 0);
 }
 #endif
 
@@ -177,10 +179,12 @@ static efi_status_t efi_init_capsule(void)
 {
 	efi_status_t ret = EFI_SUCCESS;
 
-#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
-	ret = efi_corstone1000_alloc_capsule_shared_buf();
+#if IS_ENABLED(CONFIG_EFI_HAVE_CAPSULE_SUPPORT) &&	\
+	(IS_ENABLED(CONFIG_TARGET_CORSTONE1000) ||	\
+	 IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP))
+	ret = efi_fvp_alloc_capsule_shared_buf();
 	if (ret != EFI_SUCCESS) {
-		printf("EFI: Corstone-1000: cannot allocate caspsule shared buffer\n");
+		printf("EFI: FVP: cannot allocate caspsule shared buffer\n");
 		return ret;
 	}
 #endif
@@ -296,9 +300,11 @@ efi_status_t efi_init_obj_list(void)
 	if (ret != EFI_SUCCESS)
 		goto out;
 
-#if IS_ENABLED(CONFIG_TARGET_CORSTONE1000)
+#if IS_ENABLED(CONFIG_EFI_HAVE_CAPSULE_SUPPORT) &&	\
+	(IS_ENABLED(CONFIG_TARGET_CORSTONE1000) ||	\
+	 IS_ENABLED(CONFIG_TARGET_VEXPRESS64_BASE_FVP))
 		int ffa_ret;
-		ffa_ret = efi_corstone1000_uboot_efi_started_event();
+		ffa_ret = efi_fvp_uboot_efi_started_event();
 		if (ffa_ret)
 			log_err("Failure to notify SE Proxy FW update service\n");
 		else
diff --git a/test/dm/nvmxip.c b/test/dm/nvmxip.c
index e934748eb5..89bf481f61 100644
--- a/test/dm/nvmxip.c
+++ b/test/dm/nvmxip.c
@@ -17,7 +17,7 @@
 #include <linux/bitops.h>
 #include <test/test.h>
 #include <test/ut.h>
-#include "../../drivers/mtd/nvmxip/nvmxip.h"
+#include <nvmxip.h>
 
 /* NVMXIP devices described in the device tree */
 #define SANDBOX_NVMXIP_DEVICES 2
-- 
2.25.1

